### PR TITLE
remove unused 'strict' arg

### PR DESCRIPTION
**Pull request by snopoke**
I originally noticed this because the default value of `strict` was `True` - since the `strict` arg is also used to skip cache this resulted in always skipping the cache.

This has been like this since 02913388706f6c7e9063f8289de6817a8652d57b

Quick test to double check: https://github.com/dimagi/commcare-hq/compare/sk/quickcache-skip?expand=1

### CODE REVIEW

**Review by dannyroberts — APPROVED** [review]
Wow! I wonder if this explains why couchdb is sometimes the bottle neck for submission throughput in load testing?

Is it implicit here that `get_by_username` is well-invalidated enough (on user save for example) that there's never good cause to call it without caching?

**Comment by snopoke** [comment]
My thoughts too. The cache is cleared on `save` and `delete`. I don't think we do any bulk functions on users that don't ultimately call these methods on individual objects but I can dig into that tomorrow to confirm.

**Review by pr33thi — APPROVED** [review]

**Review by snopoke — COMMENTED** [review]
Did some digging with this and found at least one place it wasn't called (picked up by tests). 

I can't find any direct uses of the bulk methods but put in a soft assert to check.

I think this is GTG once build passes.

**Comment by snopoke** [comment]
Assert is failing build and some places that call the function with `domain` as the second arg - which just seems completely incorrect.

I'll try look at these later - otherwise on Monday

**Comment by snopoke** [comment]
Still some funkiness going on, looks like some saves happening that shouldn't - I'll pick this up again next week.

File: corehq/apps/domain/shortcuts.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def create_user(username, password, is_staff=False, is_superuser=False, is_activ
     else:
         user.password = password
 
+    user.DO_NOT_SAVE_COUCH_USER = True
```
Comment (dannyroberts) [inline corehq/apps/domain/shortcuts.py]: Why is this fine to do? Is it because we already explicitly create a couch user as well?

Comment (snopoke) [reply corehq/apps/domain/shortcuts.py]: Yes. This is called as the first step in creating a new user which means that the Couch user doesn't exist yet. This isn't strictly necessary but it does save a few DB / cache hits and it makes more logical sense to me.

I'll add a comment.

**Review by dannyroberts — APPROVED** [review]

**Pull request merged**